subtests/docker_cli/dockerinspect/test_inspect_keys.py

```diff
@@ -0,0 +1,270 @@
+        "VolumeDriver": "",
+        "ShmSize": 67108864
+    },
+    "GraphDriver": {
+        "Name": "devicemapper",
+        "Data": {
+            "DeviceId": "2310",
+            "DeviceName": "docker-253:3-....",
+            "DeviceSize": "107374182400"
+        }
+    },
+    "Mounts": [],
+    "Config": {
+        "Hostname": "ed3b9325177c",
+        "Domainname": "",
+        "User": "",
+        "AttachStdin": true,
+        "AttachStdout": true,
+        "AttachStderr": true,
+        "Tty": true,
+        "OpenStdin": true,
+        "StdinOnce": true,
+        "Env": [
+            "container=docker",
+            "PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin"
+        ],
+        "Cmd": [
+            "tail",
+            "-n1",
+            "/proc/1/cgroup"
+        ],
+        "Image": "rhel7/rhel:latest",
+        "Volumes": null,
+        "WorkingDir": "",
+        "Entrypoint": null,
+        "OnBuild": null,
+        "Labels": {
+            "Architecture": "x86_64",                                 |
+            "Authoritative_Registry": "registry.access.redhat.com",   |
+            "BZComponent": "rhel-server-docker",                      |
+            "Build_Host": "rcm-img02.build.eng.bos.redhat.com",       |
+            "Name": "rhel7/rhel",                                     |
+            "Release": "46",                                          |
+            "Vendor": "Red Hat, Inc.",                                |
+            "Version": "7.2"                                          |
+        },
+        "StopSignal": "SIGTERM"
+    },
+    "NetworkSettings": {
+        "Bridge": "",
+        "SandboxID": "",
+        "HairpinMode": false,
+        "LinkLocalIPv6Address": "",
+        "LinkLocalIPv6PrefixLen": 0,
+        "Ports": null,
+        "SandboxKey": "",
+        "SecondaryIPAddresses": null,
+        "SecondaryIPv6Addresses": null,
+        "EndpointID": "",
+        "Gateway": "",
+        "GlobalIPv6Address": "",
+        "GlobalIPv6PrefixLen": 0,
+        "IPAddress": "",
+        "IPPrefixLen": 0,
+        "IPv6Gateway": "",
+        "MacAddress": "",
+        "Networks": {
+            "bridge": {                    | "RenamedForTesting_bridge": {
+                "EndpointID": "",
+                "Gateway": "",
+                "IPAddress": "",
+                "IPPrefixLen": 0,
+                "IPv6Gateway": "",
+                "GlobalIPv6Address": "",
+                "GlobalIPv6PrefixLen": 0,
+                "MacAddress": ""
+            }
+        }
+    }
+}
+]
+""")
+
+    def test_meta(self):
+        """
+        Make sure the above tests aren't just NOPing.
+        """
+        raised = False
+        try:
+            self._run_one_test("""
+[
+{
+    "This":"Should"         | "Cause": "A Test Failure"
+}
+]
+""")
+        except AssertionError:
+            raised = True
+        if not raised:
+            self.fail("Known-bad test did not raise exception.")
```

Comment:
Hehe, I think in the time you've been working on this, a few runs through my CI job would have been more time-effective.  I'll concentrate on getting that working and leave the real programming to you :grinning: